File: 1/1.py
from typing import List


def prep_elf(text: str) -> List[int]:
    calories = []
    for line in text.split("\n"):
        calories.append(int(line))
    return calories


def get_top_calories(text: str) -> int:
    raw_elves = text.split("\n\n")
    elf_counts = [prep_elf(elf) for elf in raw_elves]
    max_calories = max([sum(x) for x in elf_counts])
    return max_calories


def get_top_three_calories(text: str) -> int:
    raw_elves = text.split("\n\n")
    elf_counts = [prep_elf(elf) for elf in raw_elves]
    calories = sorted([sum(x) for x in elf_counts], reverse=True)
    return sum(calories[:3])


def part_1_tests():
    with open("1_test.txt", "r") as f:
        data = f.read()
    assert get_top_calories(data) == 24000


def part_1():
    with open("1_input.txt", "r") as f:
        data = f.read()
    print(get_top_calories(data))


def part_2_tests():
    with open("1_test.txt", "r") as f:
        data = f.read()
    assert get_top_three_calories(data) == 45000


def part_2():
    with open("1_input.txt", "r") as f:
        data = f.read()
    print(get_top_three_calories(data))


part_1_tests()
part_1()
part_2_tests()
part_2()

File: 1/1.cc
#include <algorithm>
#include <fstream>
#include <iostream>
#include <string>
#include <vector>

struct greater {
  template <class T>
  bool operator()(T const &a, T const &b) const {
    return a > b;
  }
};

int getTopCalories(std::string filename) {
  std::string buffer;
  std::ifstream InputFile(filename);
  std::vector<int> calories;
  int elfCalories = 0;
  int topCalories;

  while (getline(InputFile, buffer)) {
    if (buffer == "") {
      calories.push_back(elfCalories);
      if (elfCalories > topCalories) {
        topCalories = elfCalories;
      }
      elfCalories = 0;
    } else {
      elfCalories += std::stoi(buffer);
    }
  }

  calories.push_back(elfCalories);
  if (elfCalories > topCalories) {
    topCalories = elfCalories;
  }

  InputFile.close();
  return topCalories;
}

int getTopThreeCalories(std::string filename) {
  std::string buffer;
  std::ifstream InputFile(filename);
  std::vector<int> calories;
  int elfCalories = 0;

  while (getline(InputFile, buffer)) {
    if (buffer == "") {
      calories.push_back(elfCalories);
      elfCalories = 0;
    } else {
      elfCalories += std::stoi(buffer);
    }
  }

  calories.push_back(elfCalories);
  InputFile.close();

  std::sort(calories.begin(), calories.end(), greater());
  int topThree = 0;
  for (int i = 0; i < 3; i++) {
    topThree += calories[i];
  }

  return topThree;
}

int part1() {
  if (getTopCalories("1_test.txt") != 24000) {
    return 1;
  }
  std::cout << getTopCalories("1_input.txt") << std::endl;
  return 0;
}

int part2() {
  if (getTopThreeCalories("1_test.txt") != 45000) {
    return 1;
  }
  std::cout << getTopThreeCalories("1_input.txt") << std::endl;
  return 0;
}

int main() {
  if (part1() == 1) {
    return 1;
  }
  if (part2() == 1) {
    return 1;
  }
  return 0;
}
